### PR TITLE
Dirichlet distribution

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**
split from #1177

### CODE REVIEW

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

File: cupy/random/generator.py [hunk 1]
```diff
@@ -90,6 +90,25 @@ def binomial(self, n, p, size=None, dtype=int):
...
+        alpha = cupy.asarray(alpha)
+        if size is None:
+            size = alpha.shape
+        y = cupy.zeros(shape=size, dtype=dtype)
+        _kernels.standard_gamma_kernel(alpha, self.rk_seed, y)
+        y /= cupy.expand_dims(y.sum(axis=-1), axis=-1)
+        if size is None:
+            self.rk_seed += 1
+        else:
+            self.rk_seed += numpy.prod(size)
```
Comment (okuta) [inline cupy/random/generator.py]: Please use `cupy.core.internal.prod`.

File: cupy/random/generator.py [hunk 2]
```diff
@@ -90,6 +90,25 @@ def binomial(self, n, p, size=None, dtype=int):
...
+        .. seealso::
+            :func:`cupy.random.dirichlet` for full documentation,
+            :meth:`numpy.random.RandomState.dirichlet`
+        """
+        alpha = cupy.asarray(alpha)
+        if size is None:
+            size = alpha.shape
+        y = cupy.zeros(shape=size, dtype=dtype)
+        _kernels.standard_gamma_kernel(alpha, self.rk_seed, y)
+        y /= cupy.expand_dims(y.sum(axis=-1), axis=-1)
```
Comment (okuta) [inline cupy/random/generator.py]: How about using `keepdims`? `y /= y.sum(axis=-1, keepdims=True)`

**Comment by YoshikawaMasashi** [comment]
fixed
jenkins, test this please.

**Comment by okuta** [comment]
Please resolve conflicts.

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
flake8

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**